bundles/org.openhab.binding.velux/src/test/java/org/openhab/binding/velux/test/TestNotificationsAndDatabase.java

```diff
@@ -0,0 +1,907 @@
+import org.openhab.binding.velux.internal.bridge.slip.SCgetProduct;
+import org.openhab.binding.velux.internal.bridge.slip.SCgetProductStatus;
+import org.openhab.binding.velux.internal.bridge.slip.SCrunProductCommand;
+import org.openhab.binding.velux.internal.things.VeluxExistingProducts;
+import org.openhab.binding.velux.internal.things.VeluxKLFAPI;
+import org.openhab.binding.velux.internal.things.VeluxKLFAPI.Command;
+import org.openhab.binding.velux.internal.things.VeluxProduct;
+import org.openhab.binding.velux.internal.things.VeluxProduct.ProductBridgeIndex;
+import org.openhab.binding.velux.internal.things.VeluxProduct.ProductState;
+import org.openhab.binding.velux.internal.things.VeluxProductName;
+import org.openhab.binding.velux.internal.things.VeluxProductPosition;
+import org.openhab.binding.velux.internal.things.VeluxProductPosition.PositionType;
+import org.openhab.binding.velux.internal.things.VeluxProductType.ActuatorType;
+import org.openhab.core.library.types.PercentType;
+
+/**
+ * JUnit test suite to check the proper parsing of actuator notification packets, and to confirm that the existing
+ * products database is working correctly.
+ *
+ * @author Andrew Fiddian-Green - Initial contribution.
+ */
+@NonNullByDefault
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+public class TestNotificationsAndDatabase {
+    // validation parameters
+    private static final byte PRODUCT_INDEX_A = 6;
+    private static final byte PRODUCT_INDEX_B = 0;
+    private static final int MAIN_POSITION_A = 0xC800;
+    private static final int MAIN_POSITION_B = 0x4600;
+    private static final int VANE_POSITION_A = 0x634f;
+    private static final int TARGET_POSITION = 0xB800;
+    private static final int STATE_SOMFY = 0x2D;
+
+    private static final int UNKNOWN_POSITION = VeluxProductPosition.VPP_VELUX_UNKNOWN;
+    private static final int IGNORE_POSITION = VeluxProductPosition.VPP_VELUX_IGNORE;
+    private static final int STATE_DONE = VeluxProduct.ProductState.DONE.value;
+
+    private static final ActuatorType ACTUATOR_TYPE_SOMFY = ActuatorType.BLIND_17;
+    private static final ActuatorType ACTUATOR_TYPE_VELUX = ActuatorType.WINDOW_4_1;
+    private static final ActuatorType ACTUATOR_TYPE_UNDEF = ActuatorType.UNDEFTYPE;
+
+    // existing products database
+    private static final VeluxExistingProducts EXISTING_PRODUCTS = new VeluxExistingProducts();
+
+    private static byte[] toByteArray(String input) {
+        String[] data = input.split(" ");
+        byte[] result = new byte[data.length];
+        for (int i = 0; i < data.length; i++) {
+            result[i] = Integer.decode("0x" + data[i]).byteValue();
+        }
+        return result;
+    }
+
+    private VeluxExistingProducts getExistingProducts() {
+        return EXISTING_PRODUCTS;
+    }
+
+    /**
+     * Confirm the existing products database is initialised.
+     */
+    @Test
+    @Order(1)
```

Comment:
Curious, is the order important?

Reply:
Yes. Each test leaves the test article in a given state, that is a prerequisite for the subsequent test.